src/executePrettier/executePrettierOnBufferRange.js

```diff
@@ -12,60 +11,80 @@ const buildPrettierStylelintOptions = require('./buildPrettierStylelintOptions')
+const executePrettierEslint = (editor: TextEditor, text: string): string =>
   allowUnsafeNewFunction(() => prettierEslint(buildPrettierEslintOptions(editor, text)));
 
-const executePrettierStylelint = (editor: TextEditor, text: string) =>
+const executePrettierStylelint = (editor: TextEditor, text: string): string =>
   prettierStylelint.format(buildPrettierStylelintOptions(editor, text));
 
-const executePrettierOrIntegration = async (editor: TextEditor, text: string) => {
-  try {
-    let formatted;
-    if (shouldUseStylelint() && isCurrentScopeCssScope(editor)) {
-      formatted = await executePrettierStylelint(editor, text);
-    } else if (shouldUseEslint()) {
-      formatted = executePrettierEslint(editor, text);
-    } else {
-      formatted = executePrettier(editor, text);
-    }
+const executePrettierOrIntegration = async (editor: TextEditor, text: string, cursorOffset: number) => {
+  const nextCursorOffset: number = cursorOffset;
+  let nextText;
 
-    return formatted;
-  } catch (error) {
-    return error;
+  if (shouldUseStylelint() && isCurrentScopeCssScope(editor)) {
+    // TODO: add support for cursor position - https://github.com/hugomrdias/prettier-stylelint/issues/13
+    nextText = await executePrettierStylelint(editor, text);
+  } else if (shouldUseEslint()) {
+    // TODO: add support for cursor position - https://github.com/prettier/prettier-eslint/issues/164
+    nextText = executePrettierEslint(editor, text);
+  } else {
+    return executePrettierWithCursor(editor, text, cursorOffset);
   }
+
+  return { formatted: nextText, cursorOffset: nextCursorOffset };
 };
 
 const executePrettierOnBufferRange = async (
   editor: TextEditor,
   bufferRange: Range,
   options?: { setTextViaDiff?: boolean },
 ) => {
-  const cursorPositionPriorToFormat = editor.getCursorScreenPosition();
-  const textToTransform = editor.getTextInBufferRange(bufferRange);
-  const transformed = await executePrettierOrIntegration(editor, textToTransform);
+  // grab cursor position and file contents
+  const currentBuffer: TextBuffer = editor.getBuffer();
+  const cursorPosition: Point = editor.getCursorBufferPosition();
+  const source: string = editor.getTextInBufferRange(bufferRange);
+  const cursorOffset: number = currentBuffer.characterIndexForPosition(cursorPosition);
+  let results: Prettier$CursorResult = {
+    cursorOffset,
+    formatted: source,
+  };
 
-  const isTextUnchanged = transformed === textToTransform;
-  if (!transformed || isTextUnchanged) return;
+  // make sure we actually have text to format
+  if (!source || source.length === 0) return;
 
-  if (_.isError(transformed)) {
-    handleError({ bufferRange, editor, error: transformed });
+  try {
+    results = await executePrettierOrIntegration(editor, source, cursorOffset);
+  } catch (error) {
+    handleError({ editor, bufferRange, error });
     return;
   }
 
+  if (results.formatted === source) return;
```

Comment:
let's keep the original logic here so it's super explicit why we return early:

```js
const isTextUnchanged = results.formatted === textToTransform;
if (isTextUnchanged) return;
```

Reply:
Sounds good!